packages/netlify-cms-backend-bitbucket/src/API.js

```diff
@@ -46,7 +46,14 @@ export default class API {
     ])(req);
 
   user = () => this.request('/user');
-  hasWriteAccess = () => this.request(this.repoURL).then(res => res.ok);
+
+  hasWriteAccess = async () => {
+    const response = await this.request(this.repoURL);
+    if (response.status === 404) {
+      throw Error('Repo not found');
```

Comment:
The bitbucket backend doesn't throw if the repo is not found, so we check for an explicit 404 status on the response.